adaptdl/adaptdl/speedup.py

```diff
@@ -135,13 +145,24 @@ def __call__(self, nodes, replicas, return_local_bsz=False):
         elif self._elastic_bsz:
             max_local_bsz = np.floor(self._max_batch_size / replicas)
             min_local_bsz = np.ceil(self._init_batch_size / replicas)
-            if self._max_local_bsz is not None:
+            if (self._max_local_bsz is not None and
+                    not self._gradient_accumulation):
                 max_local_bsz = np.minimum(self._max_local_bsz, max_local_bsz)
             if self._min_local_bsz is not None:
                 min_local_bsz = np.maximum(self._min_local_bsz, min_local_bsz)
+            if self._gradient_accumulation:
+                original_min_local_bsz = min_local_bsz
+                min_local_bsz = np.where(replicas == 1,
+                                         self._max_local_bsz,
+                                         min_local_bsz)
```

Comment:
There's some shenanigans going on here, mostly with how the batchsize candidates for gradient accumulation on a single replica either need to be the initial batch size or force gradient accumulation. I think it will be easier to just rework this to be done by editing `local_bsz` after `local_bsz = np.geomspace(min_local_bsz, max_local_bsz, num=100)` and force `self._partition_local_bsz` to partition when its not the initial batch size

Reply:
Reworked this logic